src/rules/parameter-destructuring/index.ts

```diff
@@ -0,0 +1,99 @@
+            recommended: 'error',
+        },
+        hasSuggestions: true,
+        schema: [
+            {
+                type: 'object',
+                properties: {
+                    allowSingleLine: {
+                        type: 'boolean',
+                    },
+                },
+                additionalProperties: false,
+            },
+        ],
+        messages: {
+            unexpectedDestructuring: (
+                'Destructuring should not be done in the parameters. '
+                + 'Bind to a variable and destructure inside the function.'
+            ),
+        },
+    },
+    defaultOptions: [
+        {
+            allowSingleLine: true,
+        },
+    ],
+    create: context => {
+        const options = context.options[0] ?? {};
+        const allowSingleLine = options.allowSingleLine ?? true;
+
+        const sourceCode = context.getSourceCode();
+
+        return {
+            ObjectPattern: function checkObjectPattern(node): void {
+                const {parent} = node;
+
+                // Skip for destructuring in contexts unrelated to function parameters
+                if (
+                    parent?.type !== AST_NODE_TYPES.FunctionExpression
+                    && parent?.type !== AST_NODE_TYPES.FunctionDeclaration
+                    && parent?.type !== AST_NODE_TYPES.ArrowFunctionExpression
+                ) {
+                    return;
+                }
+
+                // Allow one-line destructuring when there is a single parameter
+                if (
+                    allowSingleLine
+                    && parent.params.length === 1
+                    && parent.loc.start.line === node.loc.start.line
+                    && parent.loc.start.line === node.loc.end.line
+                ) {
+                    return;
+                }
+
+                const {body} = parent;
+
+                context.report({
+                    node: node,
+                    messageId: 'unexpectedDestructuring',
+                    suggest: body.type === AST_NODE_TYPES.BlockStatement
+                        ? [
+                            {
+                                messageId: 'unexpectedDestructuring',
+                                fix: (fixer): RuleFix[] => [
+                                    // Replace the destructuring with a variable declaration
+                                    fixer.replaceText(node, 'value'),
+
+                                    // Add the destructuring inside the body
+                                    fixer.insertTextAfter(
+                                        // Null safety: a block expression always have the `{` token
+                                        // opening the body of the function.
+                                        sourceCode.getFirstToken(body)!,
+                                        `\nconst ${sourceCode.getText(node)} = value;\n`,
```

Comment:
This rule should not be auto-fixable IMHO because it introduces a meaningless parameter name that gets unnoticed. Also, fixing multiparameter is still more complex.

Reply:
It is not auto-fixable, it just provide suggestions for the IDE

Reply:
Oh, nice. I didn't know that was possible just to suggest. Pretty nice!